cmd/github-ldap-user-group-creator/main.go

```diff
@@ -240,8 +244,24 @@ func makeGroups(mapping map[string]string, roverGroups map[string][]string, conf
 	return groups, kerrors.NewAggregate(errs)
 }
 
-func ensureGroups(ctx context.Context, clients map[string]ctrlruntimeclient.Client, groupsToCreate map[string]GroupClusters, dryRun bool) error {
+func ensureGroups(ctx context.Context, clients map[string]ctrlruntimeclient.Client, groupsToCreate map[string]GroupClusters, maxConcurrency int, dryRun bool) error {
 	var errs []error
+
+	handleGroup := func(cluster string, client ctrlruntimeclient.Client, group *userv1.Group) error {
+		if err := validate(group); err != nil {
+			return fmt.Errorf("attempt to create invalid group %s on cluster %s: %w", group.Name, cluster, err)
+		}
+		logrus.WithField("cluster", cluster).WithField("group.Name", group.Name).Info("Upserting group ...")
```

Comment:
This logger can be reused on line 261. Also, you can use `WithFields` instead.

Reply:
Updated.